xdsl/xdsl_opt_main.py

```diff
@@ -250,8 +250,8 @@ def setup_pipeline(self):
             if p not in self.available_passes:
                 raise Exception(f"Unrecognized pass: {p}")
 
-        self.pipeline = [(p, lambda op, p=p: self.available_passes[p]
-                          (self.ctx, op)) for p in pipeline]
+        self.pipeline = [(p, lambda op: self.available_passes[p](self.ctx, op))
```

Comment:
I'm 99% sure that this will break.
https://stackoverflow.com/questions/19837486/lambda-in-a-loop

Reply:
Ah right, I remember this.
- [ ] Fix this